### PR TITLE
Fix prune() to remove nodes with no outgoing edges

### PR DESCRIPTION
**Pull request by Fukata-K**
This pull request addresses Issue #16.

It fixes a bug in `graph.py` where nodes with no outgoing edges (i.e., no children) could remain in the graph after calling `g.apply_topn(...)`.

### Summary of fix
In the `prune()` method, the loop exit condition now also checks whether the `in_graph` edge mask has changed, in addition to `nodes_in_graph`. This prevents premature exit of the pruning loop.

### Example
After this fix, calling:

```python
g.apply_topn(50, True)
```

in `greater_than.ipynb` correctly removes node `m4` from the graph.

Let me know if you have any questions or suggestions. Thanks!

### CODE REVIEW

File: src/eap/graph.py [hunk 1]
```diff
@@ -580,15 +580,16 @@ def prune(self):
...
             self.nodes_in_graph[:] = new_nodes_in_graph
             
             # remove edges with missing parents or children
             forward_in_graph = self.nodes_in_graph.float()
             backward_in_graph = (self.nodes_in_graph.float() @ self.forward_to_backward.float())
             backward_in_graph[-1] = 1  # logits node is always present
             edge_remask = einsum(forward_in_graph, backward_in_graph, 'forward, backward -> forward backward') > 0
+            old_edges_in_graph = self.in_graph.clone()
             self.in_graph *= edge_remask
+            old_new_same = torch.all(new_nodes_in_graph == self.nodes_in_graph) and torch.all(old_edges_in_graph == self.in_graph)
```
Comment (hannamw) [inline src/eap/graph.py]: `torch.all(new_nodes_in_graph == self.nodes_in_graph)` is guaranteed to be true, since we earlier set `self.nodes_in_graph[:] = new_nodes_in_graph`, right? So we need to check whether there were any node changes (as done in the old version), along with checking for any edges changes, when computing old_new_same.

Comment (Fukata-K) [reply src/eap/graph.py]: Thanks for the clarification! I've updated the loop condition to compare both the previous `nodes_in_graph` and `in_graph` states as suggested. Let me know if there's anything else I should adjust.
